include/souper/Infer/Interpreter.h

```diff
@@ -0,0 +1,67 @@
+#ifndef SOUPER_INTERPRTER_H
+#define SOUPER_INTERPRTER_H
+
+#include "llvm/Support/KnownBits.h"
+#include "llvm/IR/ConstantRange.h"
+
+#include "souper/Inst/Inst.h"
+
+namespace souper {
+
+struct EvalValue {
+  enum class Kind {Val, Poison, Undef, Unimplemented};
```

Comment:
I think it would be best to rename this enum since Souper already has a type called Kind that is important and pervasive